src/main/java/MorseCodePresenter.java

```diff
@@ -0,0 +1,48 @@
+public class MorseCodePresenter
+{
+    private MorseCodeFrame view;
+    private MorseCodeConverter model;
+
+    public MorseCodePresenter(MorseCodeFrame view, MorseCodeConverter model)
+    {
+        this.view = view;
+        this.model = model;
+    }
+
+    public void translateToEnglish(String expression)
+    {
+        if (view.getMorseExpressionText() == null || view.getMorseExpressionText().strip().equals(""))
```

Comment:
Everything the presenter needs should be passed into the method.
You should not call a get method on the view.

Reply:
That's a really good point - I had shifted out that statement when removing error messages but had forgotten that expression is the same thing as view.getMorseExpressionText()